lib/internal/fs/utils.js

```diff
@@ -749,7 +749,7 @@ let nonPortableTemplateWarn = true;
 function warnOnNonPortableTemplate(template) {
   // Template strings passed to the mkdtemp() family of functions should not
   // end with 'X' because they are handled inconsistently across platforms.
-  if (nonPortableTemplateWarn && StringPrototypeEndsWith(template, 'X')) {
+  if (nonPortableTemplateWarn && StringPrototypeEndsWith(`${template}`, 'X')) {
```

Comment:
`String.prototype.endsWith` always strigifies `this` value but explicit coercion wouldn't hurt

Reply:
I think it would be a mistake to attempt to stringify a TypedArray, let's not bother with this warning if the template is not a string, wdyt?
```suggestion
  if (nonPortableTemplateWarn && typeof template === 'string' && StringPrototypeEndsWith(template, 'X')) {
```

Reply:
Oh that's right, it can be native Uint8Array, too. Fixed that in the actual functions. 😅

As for warning, AFAICT there's no utf character that ends with `0x58` except for one-byte `X`, so `Array.prototype.at.call(template, -1) === 0x58` should suffice for buffers.

Reply:
I'm not sure it's an always correct assumption that the system would be using a UTF encoding set, but maybe it's fair to consider so as it's certainly very common 🤔 but probably folks who are using a `Buffer`/`Uint8Array` are doing so precisely so it can work on weird encoding sets? Maybe it's not worth overthinking it, it's just emitting a warning, so it's not like it would block anyone to have false positive.

Reply:
The warning makes sense because the underlying implementation still might get confused.
[glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/tempname.c;h=f24d962288852211b0fb79e5d22f55724b8adefa;hb=HEAD#l287) performs multibyte-unfriendly check with strspn, and it might be likely for `mkdtemp` on exotic platforms supporting prefixes longer than 6 to be as rough as `for (int i = strlen(template); template[--i] == 'X'; template[i] = getRandomChar()) {}`.
IMHO if someone writes precise code for specific platforms and charsets, it's probably up to them to simply ignore warning about template not being portable.